### PR TITLE
fix: broken rich text onChange prop

### PR DESCRIPTION
**Pull request by MayaGillilan**
Closes https://github.com/contentful/field-editors/issues/1236

### CODE REVIEW

File: packages/rich-text/src/RichTextEditor.tsx [hunk 1]
```diff
@@ -88,14 +104,19 @@ export const ConnectedRichTextEditor = (props: ConnectedProps) => {
...
+  const { sdk, isInitiallyDisabled, onAction, restrictedMarks, onChange, ...otherProps } = props;
   const isEmptyValue = React.useCallback(
     (value) => !value || deepEquals(value, Contentful.EMPTY_DOCUMENT),
     []
   );
+  // TODO: Deprecated use of onChange, should be removed in next major version
+  // To prevent breaking change, assigning onChange to SDK handler
+  React.useEffect(() => {
+    if (onChange) {
+      sdk.field.onValueChanged(onChange);
```
Comment (z0al) [inline packages/rich-text/src/RichTextEditor.tsx:117]: we should unsubscribe here on unmount

File: packages/rich-text/src/RichTextEditor.tsx [hunk 2]
```diff
@@ -88,14 +104,19 @@ export const ConnectedRichTextEditor = (props: ConnectedProps) => {
...
-
-const RichTextEditor = (props: Props) => {
-  const { sdk, isInitiallyDisabled, onAction, restrictedMarks, ...otherProps } = props;
+const RichTextEditor = (props: RichTextProps) => {
+  const { sdk, isInitiallyDisabled, onAction, restrictedMarks, onChange, ...otherProps } = props;
   const isEmptyValue = React.useCallback(
     (value) => !value || deepEquals(value, Contentful.EMPTY_DOCUMENT),
     []
   );
+  // TODO: Deprecated use of onChange, should be removed in next major version
```
Comment (z0al) [inline packages/rich-text/src/RichTextEditor.tsx:113]: Users won't see this. You should move the notice next to the prop declaration and make it a JS doc e.g.

```typescript
/** 
* @deprecated use `sdk.field.onValueChanged` instead.
*/
onChange?: ...
```

File: packages/rich-text/src/RichTextEditor.tsx (hunk 1 above)
Comment (z0al) [reply packages/rich-text/src/RichTextEditor.tsx:117]: `onValueChanged` returns an unsubscribe function already

Comment (MayaGillilan) [reply packages/rich-text/src/RichTextEditor.tsx:117]: True 😅  forgot about that

**Review by z0al — APPROVED** [review]
Thanks

**Pull request merged**